### PR TITLE
Ax 285 2 sided apt farms contracts

### PR DESCRIPTION
**Pull request by MichaelFrank0317**
# Description
Implement 2 sided apt farms so that we could stake, claim and unstake on the pools on gysr platform.

Fixes Jira Ticket # https://athletex.atlassian.net/browse/AX-285

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Is this a UI Change? If so please include screenshot of before and after states below
image
image
image
image

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have assigned 2 reviewers to check my work
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/pages/DesktopFarm.dart [hunk 1]
```diff
@@ -159,7 +155,7 @@ class _DesktopFarmState extends State<DesktopFarm> {
                     if (!isAllFarms) {
                       myController.clear();
                       setState(() {
-                        allFarmsListSearchFilter = allFarmsList;
+                        // allFarmsListSearchFilter = allFarmsList;
```
Comment (ketonkss4) [inline lib/pages/DesktopFarm.dart:158]: if this isnt needed any more just remove it

File: lib/pages/DesktopFarm.dart [hunk 2]
```diff
@@ -230,24 +227,19 @@ class _DesktopFarmState extends State<DesktopFarm> {
             mainAxisAlignment: MainAxisAlignment.spaceBetween,
             children: <Widget>[
               Text(
-                "Total APY",
+                "Current APR",
                 style: txStyle,
               ),
-              Text("12%", style: txStyle)
+              // Obx(() => Text("${farm.strAPR}%", style: txStyle))
```
Comment (ketonkss4) [inline lib/pages/DesktopFarm.dart:233]: Do not check in commented old code, just remove it if it is not needed

File: lib/pages/DesktopFarm.dart [hunk 3]
```diff
@@ -230,24 +227,19 @@ class _DesktopFarmState extends State<DesktopFarm> {
...
               Text("TVL", style: txStyle),
-              Text("\$1,000,000", style: txStyle)
-            ],
-          ),
-          Row(
-            mainAxisAlignment: MainAxisAlignment.spaceBetween,
-            children: <Widget>[
-              Text("LP APY", style: txStyle),
-              Text("5%", style: txStyle)
+              // Obx(() => Text("\$ ${farm.strTVL}", style: txStyle))
```
Comment (ketonkss4) [inline lib/pages/DesktopFarm.dart:241]: remove if not needed

File: lib/pages/DesktopFarm.dart [hunk 4]
```diff
@@ -403,31 +401,37 @@ class _DesktopFarmState extends State<DesktopFarm> {
...
+          // ),
+          // // Rewards
+          // Row(
+          //   mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          //   children: <Widget>[
+          //     Text("AX Rewards APR", style: txStyle),
+          //     Text("10%", style: txStyle)
+          //   ],
+          // ),
+
```
Comment (ketonkss4) [inline lib/pages/DesktopFarm.dart:427]: remove

File: lib/pages/DesktopFarm.dart [hunk 5]
```diff
@@ -456,18 +460,19 @@ class _DesktopFarmState extends State<DesktopFarm> {
...
-                          .where((farm) => farm.name
-                              .toUpperCase()
-                              .contains(value.toUpperCase()))
-                          .toList();
+                      // allFarmsListSearchFilter = allFarmsList
+                      //     .where((farm) => farm.strName
+                      //         .toUpperCase()
+                      //         .contains(value.toUpperCase()))
+                      //     .toList();
                     }
```
Comment (ketonkss4) [inline lib/pages/DesktopFarm.dart:474]: remove

File: lib/service/Controller/WalletController.dart [hunk 6]
```diff
@@ -58,6 +58,34 @@ class WalletController extends GetxController {
...
+      print("Raw Balance: $rawBalance");
+      EtherAmount balanceInWei = EtherAmount.inWei(rawBalance);
+      double balanceInEther = balanceInWei.getValueInUnit(EtherUnit.ether);
+      tokenBalance = balanceInEther.toStringAsFixed(6);
+    } catch (error) {
+      print("[Console] Failed to retrieve the balance: $error");
+    }
+    update();
+    return tokenBalance;
+  }
```
Comment (ketonkss4) [inline lib/service/Controller/WalletController.dart:87]: Lets put this under test

**Review by bekchanovj — APPROVED** [review]

File: lib/service/Controller/WalletController.dart [hunk 7]
```diff
@@ -58,6 +58,34 @@ class WalletController extends GetxController {
     return tokenBalance;
   }
 
+  // Update token balance
+  Future<String> getWalletBalance(
```
[the remark this answers is not in the document]
Comment (MichaelFrank0317) [reply lib/service/Controller/WalletController.dart]: It's a unnecessary function which is not used anymore. so, I removed this function.

File: lib/pages/DesktopFarm.dart (hunk 1 above)
Comment (MichaelFrank0317) [reply lib/pages/DesktopFarm.dart]: I removed it

File: lib/pages/DesktopFarm.dart (hunk 2 above)
Comment (MichaelFrank0317) [reply lib/pages/DesktopFarm.dart]: I removed it

File: lib/pages/DesktopFarm.dart (hunk 3 above)
Comment (MichaelFrank0317) [reply lib/pages/DesktopFarm.dart]: I removed it

File: lib/pages/DesktopFarm.dart (hunk 4 above)
Comment (MichaelFrank0317) [reply lib/pages/DesktopFarm.dart]: I removed it

File: lib/pages/DesktopFarm.dart (hunk 5 above)
Comment (MichaelFrank0317) [reply lib/pages/DesktopFarm.dart]: I removed it

File: lib/service/Controller/WalletController.dart (hunk 6 above)
Comment (MichaelFrank0317) [reply lib/service/Controller/WalletController.dart]: I removed it

**Review by ketonkss4 — CHANGES_REQUESTED** [review]

**Review by ketonkss4 — APPROVED** [review]

File: lib/service/GraphQL/GraphQLClientHelper.dart [hunk 8]
```diff
@@ -2,9 +2,11 @@ import 'package:flutter/cupertino.dart';
 import 'package:graphql_flutter/graphql_flutter.dart';
 
 class GraphQLClientHelper {
-  final HttpLink _httpLink = HttpLink(
-      "https://api.thegraph.com/subgraphs/name/nyamwaya/athletex-dex-subgraph-mainnet");
+  // final HttpLink _httpLink = HttpLink(
+  //     'https://thegraph.com/hosted-service/subgraph/nyamwaya/athletex-dex-subgraph-mainnet?selected=playground');
```
[the remark this answers is not in the document]
Comment (MichaelFrank0317) [reply lib/service/GraphQL/GraphQLClientHelper.dart]: I removed it.

File: lib/service/GraphQL/GraphQLConfiguration.dart [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import 'package:graphql_flutter/graphql_flutter.dart';
+
+class GraphQLConfiguration {
+  // final HttpLink _httpLink = HttpLink(
+  //     'https://thegraph.com/hosted-service/subgraph/nyamwaya/athletex-dex-subgraph-mainnet?selected=playground');
```
[the remark this answers is not in the document]
Comment (MichaelFrank0317) [reply lib/service/GraphQL/GraphQLConfiguration.dart:5]: I removed it.

File: lib/service/Controller/Farms/Farm.dart [hunk 10]
```diff
@@ -1,7 +1,162 @@
...
+        BigInt.from(100), stakingData, rewardData,
+        credentials: controller.credentials);
+    controller.updateTxString(txHash);
+  }
+
+  /// This function is used to get balance of staked farm
+  ///
+  /// @param {String} the address of the account
+  /// @return {void}
+  Future<void> getStakedBalance(String strAccount) async {
```
[the remark this answers is not in the document]
Comment (MichaelFrank0317) [reply lib/service/Controller/Farms/Farm.dart]: I updated the name of this functions as [updateStakedBalance].

**Review by Friendlygeckos — APPROVED** [review]

**Pull request merged**